### PR TITLE
Available bidders chart

### PR DESCRIPTION
**Pull request by mjoyce91**
~~Merge #1307 first~~ Merged

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
image
😍

File: src/sass/_availableBidders.scss [hunk 1]
```diff
@@ -13,6 +13,49 @@
     background-color: $bg-blue-dark-1;
   }
 
+  .available-bidders-stats {
+
+    .statistics-section {
+      background-color: $color-white;
+      border: 2px solid $color-gray-light;
+      padding: 1.2em 10% 2em 1.5em;
```
Comment (elizabeth-jimenez) [inline src/sass/_availableBidders.scss:21]: Looks much better in a box 🎊

File: src/Components/AvailableBidders/AvailableBidderStats/AvailableBidderStats.jsx [hunk 2]
```diff
@@ -1,49 +1,46 @@
...
+    { name: 'Overcompliment (OC)', key: 'OC', value: 0, color: '#112E51' },
+    { name: 'Unassigned (UA)', key: 'UA', value: 0, color: '#205493' },
+    { name: 'In Transit (IT)', key: 'IT', value: 0, color: '#9BDAF1' },
+    { name: 'Absent without leave (AWOL)', key: 'AWOL', value: 0, color: '#02BFE7' },
   ];
 
-  const COLORS = ['#102f51', '#cc3334', '#c49208', '#2970bc'];
+  data = data.map(m => ({ ...m, value: get(stats, m.key, 0) }));
+
+  let sum = sumBy(data, 'value');
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidders/AvailableBidderStats/AvailableBidderStats.jsx:28]: 👍 nice

File: src/Components/AvailableBidders/AvailableBidderStats/AvailableBidderStats.jsx [hunk 3]
```diff
@@ -58,30 +55,41 @@ const AvailableBidderStats = (props) => {
...
-            <div className="usa-width-one-whole section">
-              <h4>Status</h4>
-              <div className="usa-width-one-whole">
-                <div className="usa-width-three-fourths align-left">
+            <div className="usa-grid-full section statistics-section">
+              <h3>Available Bidders Statistics</h3>
+              <div className="usa-grid-full flex">
+                <div className="usa-width-one-fourth legend-container">
+                  <div className="usa-grid-full legend">
+                    <h4>Available Bidders by Status</h4>
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidders/AvailableBidderStats/AvailableBidderStats.jsx:63]: We might be able to do without this title

**Comment by elizabeth-jimenez** [comment]
Looks great 😍  Just some conflicts to resolve 👍

**Comment by elizabeth-jimenez** [comment]
image
Should we just hide stats section when there are no Available Bidders?

File: src/Components/LoadingText/LoadingText.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from 'react';
+
+const LinkButton = () => {
+  const [suffix, setSuffix] = useState('...');
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setSuffix(prev => prev.length < 3 ? `${prev}.` : '.');
+    }, 500);
+    return () => {
+      clearInterval(timer);
```
Comment (elizabeth-jimenez) [inline src/Components/LoadingText/LoadingText.jsx:10]: This is so cool 😎

**Comment by mjoyce91** [comment]
> image
> Should we just hide stats section when there are no Available Bidders?

Good point, that looks weird. I will update

File: src/sass/_availableBidders.scss [hunk 5]
```diff
@@ -13,6 +13,49 @@
...
+
+      h3, h4 {
+        font-family: 'Roboto';
```
Comment (elizabeth-jimenez) [inline src/sass/_availableBidders.scss:24]: 🤖

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 6]
```diff
@@ -1,96 +1,113 @@
...
-    // eslint-disable-next-line react/prop-types
-    cx, cy, midAngle, innerRadius, outerRadius, percent,
-  }) => {
-    const radius = (innerRadius + (outerRadius - innerRadius)) * 0.5;
-    const x = cx + (radius * Math.cos(-midAngle * RADIAN));
-    const y = cy + (radius * Math.sin(-midAngle * RADIAN));
+  if (sum === 0) {
+    data = data.map(m => ({ ...m, value: 1 }));
+    sum = sumBy(data, 'value');
+  }
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:37]: What is the use-case for this? 🤔

Comment (elizabeth-jimenez) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:37]: oh, i think to avoid a division by 0

Comment (mjoyce91) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:37]: yeah, plus no point displaying the chart if the sum is 0

Comment (elizabeth-jimenez) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:37]: Using `statsSum` for not displaying, right?

Comment (mjoyce91) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:37]: right

**Review by elizabeth-jimenez — APPROVED** [review]
🍨 🍬 🥧 🍰 👍

**Comment by elizabeth-jimenez** [comment]
What a beaut 😍

**Pull request merged**